### ISSUE TITLE
Lock exception on large files causes upload to fail

### ISSUE THREAD

**Issue by AdamVenn** [e-alfred/flowupload#268]
Thanks once again for a great app. I think I may be pushing it to its limit, but here we go!

### Steps to reproduce
1. Use Flow Upload to upload large file. I tried 50GB. 
2. Wait for upload to reach 100%

### Expected behaviour
The large file should finish uploading. Maybe the exception below can be handled for Flow Upload to wait for the lock to be released and try again.

### Actual behaviour
The upload fails as it finishes, logging the following error:
<details>
<summary>Error message</summary>
{"Exception":"OCP\\Lock\\LockedException",
"Message":"\"bigtest\" is locked",
"Code":0,
"Trace":[{"file":"/var/www/nextcloud/lib/private/Files/View.php",
"line":1156,
"function":"changeLock",
"class":"OC\\Files\\View",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/Files/View.php",
"line":1001,
"function":"basicOperation",
"class":"OC\\Files\\View",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/Files/Filesystem.php",
"line":735,
"function":"fopen",
"class":"OC\\Files\\View",
"type":"->"},

{"file":"/var/www/nextcloud/apps/flowupload/lib/Service/Flow/File.php",
"line":149,
"function":"fopen",
"class":"OC\\Files\\Filesystem",
"type":"::"},

{"file":"/var/www/nextcloud/apps/flowupload/lib/Controller/UploadController.php",
"line":101,
"function":"save",
"class":"OCA\\flowupload\\Service\\Flow\\File",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
"line":170,
"function":"upload",
"class":"OCA\\flowupload\\Controller\\UploadController",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
"line":100,
"function":"executeController",
"class":"OC\\AppFramework\\Http\\Dispatcher",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php",
"line":137,
"function":"dispatch",
"class":"OC\\AppFramework\\Http\\Dispatcher",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php",
"line":47,
"function":"main",
"class":"OC\\AppFramework\\App",
"type":"::"},

{"function":"__invoke",
"class":"OC\\AppFramework\\Routing\\RouteActionHandler",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/Route/Router.php",
"line":297,
"function":"call_user_func"},

{"file":"/var/www/nextcloud/lib/base.php",
"line":1010,
"function":"match",
"class":"OC\\Route\\Router",
"type":"->"},

{"file":"/var/www/nextcloud/index.php",
"line":37,
"function":"handleRequest",
"class":"OC",
"type":"::"}],
"File":"/var/www/nextcloud/lib/private/Files/View.php",
"Line":2000,
"Previous":{"Exception":"OCP\\Lock\\LockedException",
"Message":"\"files/381b82257fbbf9cabc55c86ff5ea19cc\" is locked",
"Code":0,
"Trace":[{"file":"/var/www/nextcloud/lib/private/Files/Storage/Common.php",
"line":816,
"function":"changeLock",
"class":"OC\\Lock\\DBLockingProvider",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/Files/Storage/Wrapper/Wrapper.php",
"line":616,
"function":"changeLock",
"class":"OC\\Files\\Storage\\Common",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/Files/Storage/Wrapper/Wrapper.php",
"line":616,
"function":"changeLock",
"class":"OC\\Files\\Storage\\Wrapper\\Wrapper",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/Files/View.php",
"line":1994,
"function":"changeLock",
"class":"OC\\Files\\Storage\\Wrapper\\Wrapper",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/Files/View.php",
"line":1156,
"function":"changeLock",
"class":"OC\\Files\\View",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/Files/View.php",
"line":1001,
"function":"basicOperation",
"class":"OC\\Files\\View",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/Files/Filesystem.php",
"line":735,
"function":"fopen",
"class":"OC\\Files\\View",
"type":"->"},

{"file":"/var/www/nextcloud/apps/flowupload/lib/Service/Flow/File.php",
"line":149,
"function":"fopen",
"class":"OC\\Files\\Filesystem",
"type":"::"},

{"file":"/var/www/nextcloud/apps/flowupload/lib/Controller/UploadController.php",
"line":101,
"function":"save",
"class":"OCA\\flowupload\\Service\\Flow\\File",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
"line":170,
"function":"upload",
"class":"OCA\\flowupload\\Controller\\UploadController",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
"line":100,
"function":"executeController",
"class":"OC\\AppFramework\\Http\\Dispatcher",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php",
"line":137,
"function":"dispatch",
"class":"OC\\AppFramework\\Http\\Dispatcher",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php",
"line":47,
"function":"main",
"class":"OC\\AppFramework\\App",
"type":"::"},

{"function":"__invoke",
"class":"OC\\AppFramework\\Routing\\RouteActionHandler",
"type":"->"},

{"file":"/var/www/nextcloud/lib/private/Route/Router.php",
"line":297,
"function":"call_user_func"},

{"file":"/var/www/nextcloud/lib/base.php",
"line":1010,
"function":"match",
"class":"OC\\Route\\Router",
"type":"->"},

{"file":"/var/www/nextcloud/index.php",
"line":37,
"function":"handleRequest",
"class":"OC",
"type":"::"}],
"File":"/var/www/nextcloud/lib/private/Lock/DBLockingProvider.php",
"Line":266},
"CustomMessage":"--"}
</details>

<details>
<summary>Server configuration</summary>

**Operating system:** Ubuntu 20.04.1 LTS

**Web server:** Apache/2.4.46

**Database:** mysql  Ver 15.1 Distrib 10.3.25-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2

**PHP version:** Version: 7.4.3, Memory Limit: 512 MB, Max Execution Time: 3600, Upload max size: 700 GB

**Nextcloud version:** 19.0.5 (Latest stable at time of writing)

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated from 19.0.4 today!

**Where did you install Nextcloud from:** `wget https://download.nextcloud.com/server/releases/nextcloud-18.0.4.tar.bz2`

**List of activated apps:**
<details>
<summary>App list</summary>
Enabled:
  - activity: 2.12.1
  - admin_audit: 1.9.0
  - bruteforcesettings: 2.0.1
  - cloud_federation_api: 1.2.0
  - contactsinteraction: 1.0.0
  - dav: 1.15.0
  - documentserver_community: 0.1.8
  - extract: 1.2.5
  - federatedfilesharing: 1.9.0
  - files: 1.14.0
  - files_downloadactivity: 1.8.0
  - files_external: 1.10.0
  - files_pdfviewer: 1.8.0
  - files_rightclick: 0.16.0
  - files_sharing: 1.11.0
  - files_trashbin: 1.9.0
  - files_videoplayer: 1.8.0
  - flowupload: 1.1.2
  - group_everyone: 0.1.6
  - logreader: 2.4.0
  - lookup_server_connector: 1.7.0
  - metadata: 0.12.0
  - oauth2: 1.7.0
  - onlyoffice: 6.1.0
  - password_policy: 1.9.1
  - privacy: 1.3.0
  - provisioning_api: 1.9.0
  - serverinfo: 1.9.0
  - settings: 1.1.0
  - sharebymail: 1.9.0
  - text: 3.0.1
  - theming: 1.10.0
  - twofactor_backupcodes: 1.8.0
  - updatenotification: 1.9.0
  - user_ldap: 1.9.1
  - viewer: 1.3.0
  - workflowengine: 2.1.0
Disabled:
  - accessibility
  - comments
  - encryption
  - federation
  - files_versions
  - firstrunwizard
  - nextcloud_announcements
  - notifications
  - photos
  - recommendations
  - support
  - survey_client
  - systemtags
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***",
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "19.0.5.2",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "htaccess.RewriteBase": "\/",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "OCA\\User_LDAP\\LDAPProviderFactory",
        "log_type": "syslog",
        "logfile": "",
        "loglevel": 3,
        "log.condition": {
            "apps": [
                "admin_audit"
            ]
        },
        "trashbin_retention_obligation": "1, 2",
        "ldapUserCleanupInterval": "30",
        "maintenance": false,
        "theme": "",
        "app_install_overwrite": [
            "files_frommail"
        ],
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    },
    "apps": {
        "accessibility": {
            "enabled": "no",
            "installed_version": "1.4.0",
            "types": ""
        },
        "activity": {
            "enabled": "yes",
            "installed_version": "2.12.1",
            "types": "filesystem"
        },
        "admin_audit": {
            "enabled": "yes",
            "installed_version": "1.9.0",
            "types": "logging"
        },
        "backgroundjob": {
            "lastjob": "3651"
        },
        "bruteforcesettings": {
            "enabled": "yes",
            "installed_version": "2.0.1",
            "types": ""
        },
        "calendar": {
            "enabled": "no",
            "installed_version": "2.0.3",
            "types": ""
        },
        "cloud_federation_api": {
            "enabled": "yes",
            "installed_version": "1.2.0",
            "types": "filesystem"
        },
        "comments": {
            "enabled": "no",
            "installed_version": "1.8.0",
            "types": "logging"
        },
        "contactsinteraction": {
            "enabled": "yes",
            "installed_version": "1.0.0",
            "types": "dav"
        },
        "core": {
            "backgroundjobs_mode": "ajax",
            "installed.bundles": "[\"CoreBundle\"]",
            "installedat": "1589901242.7554",
            "lastcron": "1606425363",
            "lastupdateResult": "[]",
            "lastupdatedat": "1606423576",
            "oc.integritycheck.checker": "[]",
            "public_files": "files_sharing\/public.php",
            "public_webdav": "dav\/appinfo\/v1\/publicwebdav.php",
            "theming.variables": "b494f9fe4e205ef1e2a61df9ff5f797d",
            "updater.secret.created": "1606413618",
            "vendor": "nextcloud"
        },
        "dav": {
            "buildCalendarReminderIndex": "yes",
            "buildCalendarSearchIndex": "yes",
            "chunks_migrated": "1",
            "enabled": "yes",
            "generateBirthdayCalendar": "no",
            "installed_version": "1.15.0",
            "regeneratedBirthdayCalendarsForYearFix": "yes",
            "sendEventReminders": "no",
            "sendInvitations": "no",
            "types": "filesystem"
        },
        "documentserver_community": {
            "enabled": "yes",
            "installed_version": "0.1.8",
            "types": "filesystem"
        },
        "extract": {
            "enabled": "yes",
            "installed_version": "1.2.5",
            "types": ""
        },
        "federatedfilesharing": {
            "enabled": "yes",
            "installed_version": "1.9.0",
            "types": ""
        },
        "federation": {
            "enabled": "no",
            "installed_version": "1.8.0",
            "types": "authentication"
        },
        "files": {
            "cronjob_scan_files": "500",
            "enabled": "yes",
            "installed_version": "1.14.0",
            "types": "filesystem"
        },
        "files_antivirus": {
            "av_cmd_options": "",
            "av_host": "",
            "av_infected_action": "delete",
            "av_max_file_size": "-1",
            "av_mode": "socket",
            "av_path": "\/usr\/bin\/clamscan",
            "av_port": "0",
            "av_socket": "\/var\/run\/clamav\/clamd.ctl",
            "av_stream_max_length": "26214400",
            "enabled": "no",
            "installed_version": "2.4.1",
            "types": "filesystem,dav"
        },
        "files_downloadactivity": {
            "enabled": "yes",
            "installed_version": "1.8.0",
            "types": "filesystem"
        },
        "files_external": {
            "enabled": "yes",
            "installed_version": "1.10.0",
            "types": "filesystem"
        },
        "files_pdfviewer": {
            "enabled": "yes",
            "installed_version": "1.8.0",
            "types": ""
        },
        "files_rightclick": {
            "enabled": "yes",
            "installed_version": "0.16.0",
            "types": ""
        },
        "files_sharing": {
            "enabled": "yes",
            "incoming_server2server_share_enabled": "no",
            "installed_version": "1.11.0",
            "lookupServerEnabled": "no",
            "lookupServerUploadEnabled": "no",
            "outgoing_server2server_share_enabled": "yes",
            "types": "filesystem"
        },
        "files_trashbin": {
            "enabled": "yes",
            "installed_version": "1.9.0",
            "types": "filesystem,dav"
        },
        "files_versions": {
            "enabled": "no",
            "installed_version": "1.11.0",
            "types": "filesystem,dav"
        },
        "files_videoplayer": {
            "enabled": "yes",
            "installed_version": "1.8.0",
            "types": ""
        },
        "firstrunwizard": {
            "enabled": "no",
            "installed_version": "2.7.0",
            "types": "logging"
        },
        "flowupload": {
            "enabled": "yes",
            "installed_version": "1.1.2",
            "types": ""
        },
        "group_everyone": {
            "enabled": "yes",
            "installed_version": "0.1.6",
            "types": "authentication"
        },
        "logreader": {
            "enabled": "yes",
            "installed_version": "2.4.0",
            "types": ""
        },
        "lookup_server_connector": {
            "enabled": "yes",
            "installed_version": "1.7.0",
            "types": "authentication"
        },
        "metadata": {
            "enabled": "yes",
            "installed_version": "0.12.0",
            "types": ""
        },
        "nextcloud_announcements": {
            "enabled": "no",
            "installed_version": "1.7.0",
            "types": "logging"
        },
        "notifications": {
            "enabled": "no",
            "installed_version": "2.6.0",
            "types": "logging"
        },
        "oauth2": {
            "enabled": "yes",
            "installed_version": "1.7.0",
            "types": "authentication"
        },
        "onlyoffice": {
            "DocumentServerUrl": "***REMOVED SENSITIVE VALUE***/index.php\/apps\/documentserver_community\/",
            "defFormats": "{\"docx\":true,\"pptx\":true,\"xlsx\":true,\"odp\":true,\"ods\":true,\"odt\":true,\"doc\":true,\"ppt\":true,\"xls\":true}",
            "editFormats": "{\"csv\":true,\"docx\":true,\"pptx\":true,\"txt\":true,\"xlsx\":true,\"odp\":true,\"ods\":true,\"odt\":true,\"rtf\":true}",
            "enabled": "yes",
            "installed_version": "6.1.0",
            "sameTab": "true",
            "types": "filesystem"
        },
        "password_policy": {
            "enabled": "yes",
            "enforceHaveIBeenPwned": "1",
            "installed_version": "1.9.1",
            "types": "authentication"
        },
        "photos": {
            "enabled": "no",
            "installed_version": "1.0.0",
            "types": ""
        },
        "privacy": {
            "enabled": "yes",
            "installed_version": "1.3.0",
            "types": ""
        },
        "provisioning_api": {
            "enabled": "yes",
            "installed_version": "1.9.0",
            "types": "prevent_group_restriction"
        },
        "recommendations": {
            "enabled": "no",
            "installed_version": "0.6.0",
            "types": ""
        },
        "serverinfo": {
            "enabled": "yes",
            "installed_version": "1.9.0",
            "types": ""
        },
        "settings": {
            "enabled": "yes",
            "installed_version": "1.1.0",
            "types": ""
        },
        "sharebymail": {
            "enabled": "yes",
            "installed_version": "1.9.0",
            "types": "filesystem"
        },
        "support": {
            "enabled": "no",
            "installed_version": "1.1.0",
            "types": "session"
        },
        "survey_client": {
            "enabled": "no",
            "installed_version": "1.6.0",
            "types": ""
        },
        "systemtags": {
            "enabled": "no",
            "installed_version": "1.8.0",
            "types": "logging"
        },
        "text": {
            "enabled": "yes",
            "installed_version": "3.0.1",
            "types": "dav"
        },
        "theming": {
            "backgroundMime": "image\/png",
            "cachebuster": "7",
            "color": "#002F7F",
            "enabled": "yes",
            "faviconMime": "image\/png",
            "installed_version": "1.10.0",
            "logoMime": "image\/png",
            "name": "***REMOVED SENSITIVE VALUE***",
            "slogan": "***REMOVED SENSITIVE VALUE***",
            "types": "logging",
            "url": "***REMOVED SENSITIVE VALUE***"
        },
        "twofactor_backupcodes": {
            "enabled": "yes",
            "installed_version": "1.8.0",
            "types": ""
        },
        "updatenotification": {
            "bruteforcesettings": "2.0.1",
            "core": "19.0.2.2",
            "documentserver_community": "0.1.7",
            "enabled": "yes",
            "extract": "1.2.5",
            "installed_version": "1.9.0",
            "metadata": "0.12.0",
            "onlyoffice": "6.0.0",
            "types": "",
            "update_check_errors": "0"
        },
        "user_ldap": {
            "cleanUpJobOffset": "0",
            "enabled": "yes",
            "installed_version": "1.9.1",
            "s01_lastChange": "1606413302",
            "s01has_memberof_filter_support": "1",
            "s01home_folder_naming_rule": "",
            "s01last_jpegPhoto_lookup": "0",
            "s01ldap_agent_password": "***REMOVED SENSITIVE VALUE***",
            "s01ldap_attributes_for_group_search": "",
            "s01ldap_attributes_for_user_search": "",
            "s01ldap_backup_host": "",
            "s01ldap_backup_port": "",
            "s01ldap_base": "***REMOVED SENSITIVE VALUE***",
            "s01ldap_base_groups": "***REMOVED SENSITIVE VALUE***",
            "s01ldap_base_users": "***REMOVED SENSITIVE VALUE***",
            "s01ldap_cache_ttl": "600",
            "s01ldap_configuration_active": "1",
            "s01ldap_default_ppolicy_dn": "",
            "s01ldap_display_name": "displayname",
            "s01ldap_dn": "***REMOVED SENSITIVE VALUE***",
            "s01ldap_dynamic_group_member_url": "",
            "s01ldap_email_attr": "",
            "s01ldap_experienced_admin": "0",
            "s01ldap_expert_username_attr": "",
            "s01ldap_expert_uuid_group_attr": "",
            "s01ldap_expert_uuid_user_attr": "",
            "s01ldap_ext_storage_home_attribute": "",
            "s01ldap_gid_number": "gidNumber",
            "s01ldap_group_display_name": "cn",
            "s01ldap_group_filter": "***REMOVED SENSITIVE VALUE***",
            "s01ldap_group_filter_mode": "0",
            "s01ldap_group_member_assoc_attribute": "member",
            "s01ldap_groupfilter_groups": "***REMOVED SENSITIVE VALUE***",
            "s01ldap_groupfilter_objectclass": "group",
            "s01ldap_host": "***REMOVED SENSITIVE VALUE***",
            "s01ldap_login_filter": "***REMOVED SENSITIVE VALUE***",
            "s01ldap_login_filter_mode": "0",
            "s01ldap_loginfilter_attributes": "",
            "s01ldap_loginfilter_email": "0",
            "s01ldap_loginfilter_username": "1",
            "s01ldap_nested_groups": "0",
            "s01ldap_override_main_server": "",
            "s01ldap_paging_size": "500",
            "s01ldap_port": "389",
            "s01ldap_quota_attr": "",
            "s01ldap_quota_def": "",
            "s01ldap_tls": "0",
            "s01ldap_turn_off_cert_check": "0",
            "s01ldap_turn_on_pwd_change": "0",
            "s01ldap_user_avatar_rule": "default",
            "s01ldap_user_display_name_2": "",
            "s01ldap_user_filter_mode": "0",
            "s01ldap_userfilter_groups": "***REMOVED SENSITIVE VALUE***",
            "s01ldap_userfilter_objectclass": "person",
            "s01ldap_userlist_filter": "***REMOVED SENSITIVE VALUE***",
            "s01use_memberof_to_detect_membership": "1",
            "types": "authentication"
        },
        "viewer": {
            "enabled": "yes",
            "installed_version": "1.3.0",
            "types": ""
        },
        "workflowengine": {
            "enabled": "yes",
            "installed_version": "2.1.0",
            "types": "filesystem"
        }
    }
}
```
</details>

**Are you using external storage, if yes which one:** local, but have not yet tried uploading to it using Flow Upload

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LDAP, connecting to a Samba 4 AD

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
+-------------------------------+-------------+
| Configuration                 | s01         |
+-------------------------------+-------------+
| hasMemberOfFilterSupport      | 1           |
| homeFolderNamingRule          |             |
| lastJpegPhotoLookup           | 0           |
| ldapAgentName                 | ***         |
| ldapAgentPassword             | ***         |
| ldapAttributesForGroupSearch  |             |
| ldapAttributesForUserSearch   |             |
| ldapBackupHost                |             |
| ldapBackupPort                |             |
| ldapBase                      | ***         |
| ldapBaseGroups                | ***         |
| ldapBaseUsers                 | ***         |
| ldapCacheTTL                  | 600         |
| ldapConfigurationActive       | 1           |
| ldapDefaultPPolicyDN          |             |
| ldapDynamicGroupMemberURL     |             |
| ldapEmailAttribute            |             |
| ldapExperiencedAdmin          | 0           |
| ldapExpertUUIDGroupAttr       |             |
| ldapExpertUUIDUserAttr        |             |
| ldapExpertUsernameAttr        |             |
| ldapExtStorageHomeAttribute   |             |
| ldapGidNumber                 | gidNumber   |
| ldapGroupDisplayName          | cn          |
| ldapGroupFilter               | ***         |
| ldapGroupFilterGroups         | ***         |
| ldapGroupFilterMode           | 0           |
| ldapGroupFilterObjectclass    | group       |
| ldapGroupMemberAssocAttr      | member      |
| ldapHost                      | ***         |
| ldapIgnoreNamingRules         |             |
| ldapLoginFilter               | ***         |
| ldapLoginFilterAttributes     |             |
| ldapLoginFilterEmail          | 0           |
| ldapLoginFilterMode           | 0           |
| ldapLoginFilterUsername       | 1           |
| ldapMatchingRuleInChainState  | unknown     |
| ldapNestedGroups              | 0           |
| ldapOverrideMainServer        |             |
| ldapPagingSize                | 500         |
| ldapPort                      | 389         |
| ldapQuotaAttribute            |             |
| ldapQuotaDefault              |             |
| ldapTLS                       | 0           |
| ldapUserAvatarRule            | default     |
| ldapUserDisplayName           | displayname |
| ldapUserDisplayName2          |             |
| ldapUserFilter                | ***         |
| ldapUserFilterGroups          | ***         |
| ldapUserFilterMode            | 0           |
| ldapUserFilterObjectclass     | person      |
| ldapUuidGroupAttribute        | auto        |
| ldapUuidUserAttribute         | auto        |
| turnOffCertCheck              | 0           |
| turnOnPasswordChange          | 0           |
| useMemberOfToDetectMembership | 1           |
+-------------------------------+-------------+
```
</details>

### Client configuration
**Browser:** Firefox 83

**Operating system:** Pop OS

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Available on request :)
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Available on request :)
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Available on request :)
```
</details>

</details>

**Comment by AdamVenn** [comment]
I believe this is the line that needs to be enclosed in a try-catch block, since it appears that the Nextcloud fopen function can throw, rather than returning false as seems to be expected:
https://github.com/e-alfred/flowupload/blob/08a98162d1a1f83d41908befc4c85e747f6b804d/lib/Service/Flow/File.php#L149

**Comment by e-alfred** [comment]
File locking problems usually happens if you don't have transactional locking enabled. Did you set up Redis for it? I couldn't find it in the config.php file you posted.

Here is the part of the manual for this: https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/files_locking_transactional.html

**Comment by JonathanTreffler** [comment]
> I believe this is the line that needs to be enclosed in a try-catch block

Well if the fopen function fails the upload has to fail, because the app will not be able to assemble the chunks.
When we developed this we assumed it would return false if it wouldn't work, but
```php
throw new FileOpenException('failed to open destination file: '.$destination);
````
doesn't actually do anything but return a error 404 on the request, as you can see here: https://github.com/e-alfred/flowupload/blob/08a98162d1a1f83d41908befc4c85e747f6b804d/lib/Controller/Errors.php#L20

We can certainly add a try-catch block, but it will not fix the error.
It would just change how the error would be logged.
Potentially the error logging would get worse, because it would maybe no longer log the stacktrace of the error.

**Comment by JonathanTreffler** [comment]
As far as i know we use the standard API for File Locking, so i am surprised you don't have the same problem in other apps.

Also i am confused why the size of the file changes the result. Maybe some filesystem limitation 🤷‍♂️ ?

**Comment by AdamVenn** [comment]
Thanks for the speedy replies, @e-alfred and @JonathanTreffler! I will look into Redis and test further.

**Comment by AdamVenn** [comment]
Success! I removed APCU memcache and installed Redis (on unix socket). I also upgraded to Nextcloud 20. Then, I tried a 40GB upload. The upload ran up to 100% and took a big pause (10 minutes, maybe?). I thought it had failed, but left the page open and went off to do something else. Came back and the upload had finished.

Moral of the story - use Redis and be very patient for large uploads. I would suggest adding something in the documentation/interface to this effect but will close the issue and leave that up to your judgement.

Thanks a lot, Flow Upload team! You rock!
